### PR TITLE
[SAI-PTF] Fix SAI cases scanner

### PR DESCRIPTION
**Pull request by zhiyuan0112**
Signed-off-by: zhiyuanliang <t-zhliang@microsoft.com>

### Description of PR

- Fix SAI header scanning code
- Remove the extra operations in Kusto Explorer

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
Fix SAI header scanner and remove the extra operations in Kusto Explorer.

#### How did you do it?
- Modify SAI header scanning code.
- Drop `CaseInvocationReport` table and integrate attributes into `CaseInvocationStaging`, so the extra operations in the Kusto Explorer can be omitted.

#### How did you verify/test it?
Tested locally.

### CODE REVIEW

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Failed<br/>- hook id: trailing-whitespace<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing test_reporting/sai_coverage/data_model/test_invocation.py<br/><br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing test_reporting/requirements.txt<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing test_reporting/requirements.txt<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>test_reporting/sai_coverage/case_scanner.py:231:15: E251 unexpected spaces around keyword / parameter equals<br/>test_reporting/sai_coverage/case_scanner.py:231:17: E251 unexpected spaces around keyword / parameter equals<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing test_reporting/requirements.txt<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>test_reporting/sai_coverage/case_scanner.py:231:15: E251 unexpected spaces around keyword / parameter equals<br/>test_reporting/sai_coverage/case_scanner.py:231:17: E251 unexpected spaces around keyword / parameter equals<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>test_reporting/sai_coverage/case_scanner.py:231:15: E251 unexpected spaces around keyword / parameter equals<br/>test_reporting/sai_coverage/case_scanner.py:231:17: E251 unexpected spaces around keyword / parameter equals<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

File: test_reporting/sai_coverage/data_model/sai_interface_header.py [hunk 1]
```diff
@@ -3,12 +3,8 @@ class SAIInterfaceHeader(object):
...
-        sai_method_table_split = intf_groupalias.split('_')
-        sai_feature = sai_method_table_split[1: len(sai_method_table_split)-2]
-
-        self.sai_header = file_name
-        self.sai_id = intf_groupname
-        self.sai_method_table = intf_groupalias
-        self.sai_feature = ''.join(sai_feature)
-        self.sai_api = intf_alias[:-10] if 'attribute' in intf_alias else intf_alias
-        self.sai_alias = intf_alias
+        self.intf_groupname = intf_groupname
```
Comment (richardyu-ms) [inline test_reporting/sai_coverage/data_model/sai_interface_header.py:6]: add doc for that those parameters are used for

File: test_reporting/sai_coverage/data_model/test_invocation.py [hunk 2]
```diff
@@ -3,17 +3,23 @@ class TestInvocation(object):
     Structure of invocation
     """
 
-    def __init__(self, file_name, class_name, case_name, case_invoc, sai_header, saiintf_id,
+    def __init__(self, id, file_name, class_name, case_name, case_invoc, sai_header, saiintf_id,
```
Comment (richardyu-ms) [inline test_reporting/sai_coverage/data_model/test_invocation.py:6]: add doc for that those parameters are used for

File: test_reporting/sai_coverage/case_scanner.py [hunk 3]
```diff
@@ -236,6 +243,7 @@ def construct_invocation_data(self, file_name, class_name, method_name, sai_inte
             saiintf_name=header_data[header_key]["intf_name"],
             saiintf_alias=header_data[header_key]["intf_alias"],
             test_set=test_set,
+            # TODO: test_platform is unknown during static scanning, and will be obtained in runtime scanning.
             test_platform="vms3-t1-dx010-1" if runnable else "",
```
Comment (richardyu-ms) [inline test_reporting/sai_coverage/case_scanner.py:247]: change this to static, this value should not depends on runnable

**Review by richardyu-ms — DISMISSED** [review]

**Review by richardyu-ms — APPROVED** [review]

**Pull request merged**